### PR TITLE
add `pkgconf` dependency for `ICU` and add patch to avoid trouble with long path names for nodejs-16.15.1/GCCcore-11.3.0

### PR DESCRIPTION
**Pull request by crubb**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/n/nodejs/nodejs-16.15.1-GCCcore-11.3.0.eb [hunk 1]
```diff
@@ -14,10 +14,18 @@ toolchain = {'name': 'GCCcore', 'version': '11.3.0'}
...
+]
+checksums = [
+    {'node-v16.15.1.tar.gz': '308aee7149c4092a53c87c28ef49e23a8d1606119e79ae68333062e2a1f94208'},
+    {'nodejs-16.15.1_fix-argument-list-too-long.patch':
+     '36b5627e60fd2b861ce3dc5445b137b4fe7ec30a4f866ea540ec56bda9686405'},
+]
 
 builddependencies = [
     ('binutils', '2.38'),
+    ('pkg-config', '0.29.2'),
```
Comment (smoors) [inline easybuild/easyconfigs/n/nodejs/nodejs-16.15.1-GCCcore-11.3.0.eb:28]: `pkg-config` is no longer maintained, so from `2022a` onward we prefer to use `pkgconf` where possible

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen2

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @crubb!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3136.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/98d336c12b677ca7ca270940fce1bcf6 for a full test report.

**Pull request merged**